File: src/msa/web-service/portalapi/src/main/java/vn/com/kns/portalapi/application/service/shop/shopImportDetail/dto/ShopImportDetailDto.java
package vn.com.kns.portalapi.application.service.shop.shopImportDetail.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import vn.com.kns.portalapi.application.service.shop.shopImport.dto.ShopImportMinDto;
import vn.com.kns.portalapi.application.service.shop.shop.dto.ShopDto;
import vn.com.kns.portalapi.application.service.supplier.dto.SupplierDto;
import vn.com.kns.portalapi.application.service.supply.dto.SupplyDto;
import vn.com.kns.portalapi.core.model.BaseEntityDto;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ShopImportDetailDto extends BaseEntityDto {
    private Integer quantity;
    private Long price;

    private ShopImportMinDto shopImport;
    private Long shopImportId;

    private SupplyDto supply;
    private Long supplyId;

    private ShopDto shop;
    private Long shopId;

    private SupplierDto supplier;
    private Long supplierId;
}

File: src/msa/main-service/authw-service/src/app/shared/services/shop/shop-import-detail/dto/shop-import-detail-dto.ts
import {ShopImportDto} from '../../shop-import/dto/shop-import-dto';
import {ShopDto} from '../../../shop/shop/dto/shop-dto';
import {SupplierDto} from '../../../supplier/dto/supplier-dto';
import {SupplyDto} from '../../../category/supply/dto/supply-dto';

export class ShopImportDetailDto {
  id: number | null;
  quantity: number;
  price: number;
  shopImportId: number;
  shopImport: ShopImportDto = new ShopImportDto();
  supplyId: number;
  supply: SupplyDto = new SupplyDto();
  shopId: number;
  shop: ShopDto = new ShopDto();
  supplierId: number;
  supplier: SupplierDto = new SupplierDto();

  constructor(quantity?: number, price?: number, supplyId?: number, shopId?: number, supplierId?: number,) {
    this.quantity = quantity;
    this.price = price;
    this.supplyId = supplyId;
    this.shopId = shopId;
    this.supplierId = supplierId;
  }
}
